ddtrace/contrib/botocore/services/sqs.py

```diff
@@ -133,52 +133,54 @@ def patched_sqs_api_call(original_func, instance, args, kwargs, function_vars):
     else:
         call_name = trace_operation
 
-    if should_instrument:
-        with core.context_with_data(
-            "botocore.patched_sqs_api_call",
-            parent=parent_ctx,
-            span_name=call_name,
-            service=schematize_service_name("{}.{}".format(pin.service, endpoint_name)),
-            span_type=SpanTypes.HTTP,
-            activate=True,
-            instance=instance,
-            args=args,
-            params=params,
-            endpoint_name=endpoint_name,
-            operation=operation,
-            call_trace=False,
-            call_key="instrumented_sqs_call",
-            pin=pin,
-        ) as ctx, ctx.get_item(ctx.get_item("call_key")):
-            core.dispatch("botocore.patched_sqs_api_call.started", [ctx])
-
-            if should_update_messages:
-                update_messages(ctx, endpoint_service=endpoint_name)
-
-            try:
-                if not func_has_run:
-                    core.dispatch(f"botocore.{endpoint_name}.{operation}.pre", [params])
-                    result = original_func(*args, **kwargs)
-                    core.dispatch(f"botocore.{endpoint_name}.{operation}.post", [params, result])
-
-                core.dispatch("botocore.patched_sqs_api_call.success", [ctx, result])
-
-                if func_run_err:
-                    raise func_run_err
-                return result
-            except botocore.exceptions.ClientError as e:
-                core.dispatch(
-                    "botocore.patched_sqs_api_call.exception",
-                    [
-                        ctx,
-                        e.response,
-                        botocore.exceptions.ClientError,
-                        config.botocore.operations[ctx[ctx["call_key"]].resource].is_error_code,
-                    ],
-                )
-                raise
+    try:
```

Comment:
Should we just be doing `with parent_ctx:` ?

Or maybe even do that around the whole function?

```python
def wrap_with_parent_ctx(name):
    def _outer(func):
        def _inner(*args, **kwargs):
            with core.ExecutionContext(name) as parent_ctx:
                return func(parent_ctx, *args, **kwargs)
        return _inner
    return _outer


@wrap_with_parent_ctx("botocore.patched_sqs_api_call.propagated")
def patched_sqs_api_call(parent_ctx, original_func, instance, args, kwargs, function_vars):
    ...
```


I don't _love_ this suggestion, just trying to think about how do we get the safety without adding to the code complexity (# levels of indentation/branches/etc)